mirage/client/dns_client_mirage.ml

```diff
@@ -3,64 +3,69 @@ open Lwt.Infix
 let src = Logs.Src.create "dns_client_mirage" ~doc:"effectful DNS client layer"
 module Log = (val Logs.src_log src : Logs.LOG)
 
-module Make (R : Mirage_random.S) (C : Mirage_clock.MCLOCK) (S : Mirage_stack.V4) = struct
+module Make (R : Mirage_random.S) (T : Mirage_time.S) (C : Mirage_clock.MCLOCK) (S : Mirage_stack.V4) = struct
 
   module Transport : Dns_client.S
-    with type flow = S.TCPV4.flow
-     and type stack = S.t
+    with type stack = S.t
      and type +'a io = 'a Lwt.t
      and type io_addr = Ipaddr.V4.t * int = struct
-    type flow = S.TCPV4.flow
     type stack = S.t
     type io_addr = Ipaddr.V4.t * int
     type ns_addr = [`TCP | `UDP] * io_addr
     type +'a io = 'a Lwt.t
     type t = {
-      rng : (int -> Cstruct.t) ;
       nameserver : ns_addr ;
+      timeout_ns : int64 ;
```

Comment:
`_ns` is kind of an overloaded abbreviation here since it could mean `nanoseconds` or `nameserver`, not sure if it matters; the mlistrings for `create` explain this.

Reply:
We are both ok with that. Since nameserver is spelled out, there should be no confusion.